catroid/src/main/java/org/catrobat/catroid/ui/ProjectActivity.java

```diff
@@ -433,4 +436,38 @@ private void showLegoSensorConfigInfo() {
 			dialog.show(getSupportFragmentManager(), LegoSensorConfigInfoDialog.DIALOG_FRAGMENT_TAG);
 		}
 	}
+
+	private void checkIfSpriteNameEqualBackground() {
+		List<Sprite> spriteList =
+				new ArrayList<>(ProjectManager.getInstance().getCurrentProject().getSpriteListWithClones());
+		// Starting from 1, because real background sprite has index = 0
+		for (int sprite = 1; sprite < spriteList.size(); ++sprite) {
+			if (spriteList.get(sprite).getName().matches("[\\s]*" + getString(R.string.background) +
+					"[\\s]*")) {
+				setNewSpriteName(sprite);
+				return;
+			}
+		}
+	}
+
+	private void setNewSpriteName(int sprite) {
+		List<Sprite> spriteList =
+				new ArrayList<>(ProjectManager.getInstance().getCurrentProject().getSpriteListWithClones());
+		String backgroundRegex = getString(R.string.background) + " \\([0-9]+\\)";
+		int maxValue = 0;
+
+		for (int spriteItr = 1; spriteItr < spriteList.size(); ++spriteItr) {
+			if (spriteList.get(spriteItr).getName().matches(backgroundRegex)) {
+				String tokens[] = spriteList.get(spriteItr).getName().split(" ");
+				int value = Integer.parseInt(tokens[1].replaceAll("[^0-9]", ""));
+				// Get biggest number in "()", because name must be unique
```

Comment:
Please remove the comment

Reply:
Hi, thanks for your improvements. Because I messed up rebasing unfortunately, I had to create a new branch and make new pull request with your changes: https://github.com/Catrobat/Catroid/pull/4356/commits.